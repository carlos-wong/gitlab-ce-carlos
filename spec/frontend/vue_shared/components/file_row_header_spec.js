import { shallowMount } from '@vue/test-utils';
import FileRowHeader from '~/vue_shared/components/file_row_header.vue';

describe('File row header component', () => {
  let vm;

  function createComponent(path) {
    vm = shallowMount(FileRowHeader, {
      propsData: {
        path,
      },
    });
  }

  afterEach(() => {
    vm.destroy();
  });

  it('renders file path', () => {
    createComponent('app/assets');

    expect(vm.element).toMatchSnapshot();
  });

  it('trucates path after 40 characters', () => {
    createComponent('app/assets/javascripts/merge_requests');

    expect(vm.element).toMatchSnapshot();
  });

  it('adds multiple ellipsises after 40 characters', () => {
    createComponent('app/assets/javascripts/merge_requests/widget/diffs/notes');

    expect(vm.element).toMatchSnapshot();
  });
});
