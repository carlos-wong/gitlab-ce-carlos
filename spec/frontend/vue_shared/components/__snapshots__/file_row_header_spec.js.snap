// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`File row header component adds multiple ellipsises after 40 characters 1`] = `
<div
  class="file-row-header bg-white sticky-top p-2 js-file-row-header"
>
  <span
    class="bold"
  >
    app/assets/javascripts/…/…/diffs/notes
  </span>
</div>
`;

exports[`File row header component renders file path 1`] = `
<div
  class="file-row-header bg-white sticky-top p-2 js-file-row-header"
>
  <span
    class="bold"
  >
    app/assets
  </span>
</div>
`;

exports[`File row header component trucates path after 40 characters 1`] = `
<div
  class="file-row-header bg-white sticky-top p-2 js-file-row-header"
>
  <span
    class="bold"
  >
    app/assets/javascripts/merge_requests
  </span>
</div>
`;
